### ISSUE TITLE
Crash When Switching Between User Defined Presets

### ISSUE THREAD

**Issue by Threeinone** [flightlessmango/MangoHud#1163]
I'm testing using user defined presets with `presets.conf` on vkcube. I don't know a lot of C(++) but I think this is a segfault?

**List relevant hardware/software information**
- Linux Distribution: Fedora 38 (cinnamon)
- Kernel Version: 6.5.8-200.fc38.x86_64
- MangoHud version: 0.7.0
- GPU: NVIDIA RTX 2070 SUPER

**To Reproduce**
~/.config/MangoHud/presets.conf is written as so:
```
[preset 1]
gpu_stats
cpu_stats
fps
frametime
throttling_status
frame_timing
text_outline

[preset 2]
gpu_stats
cpu_stats
core_load
vram
ram
fps
frametime
throttling_status
frame_timing
text_outline
```

standard mangohud conf
then while testing with vkcube, using `SHIFT_R+F10`, it crashes and I get and error message.

```
Selected GPU 0: NVIDIA GeForce RTX 2070 SUPER, type: DiscreteGpu
/usr/include/c++/13/bits/basic_string.h:1305: std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::front() [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; reference = char&]: Assertion '!empty()' failed.
[1]    62313 IOT instruction (core dumped)  mangohud vkcube
```

using `mangohud gdb vkcube`, I got this error message

```
Thread 1 "vkcube" received signal SIGABRT, Aborted.
__pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
44            return INTERNAL_SYSCALL_ERROR_P (ret) ? INTERNAL_SYSCALL_ERRNO (ret) : 0;                                                           
Missing separate debuginfos, use: dnf debuginfo-install xorg-x11-drv-nvidia-libs-535.113.01-1.fc38.x86_64
```

**Expected behavior**
override existing presets with a simple and detailed readout

**Additional context**
I may be stupid and just wrote the config wrong, but if that's the case it wasn't clear with existing error messages

**Comment by mc711** [comment]
i encountered this problem recently.

problem is related to parsing empty lines even though the preset sample contains empty lines...

solution: remove empty lines between presets

**Comment by flightlessmango** [comment]
should be fixed here 50d578973c4428b953d2b37de4843df0a8e691d2
